javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -109,7 +109,8 @@
     @SafeVarargs
     @SuppressWarnings("varargs")
     public static <T> Vector<T> of(T... elements) {
-        return ofAll(Iterator.of(elements));
+        Objects.requireNonNull(elements, "elements is null");
```

Comment:
should be optimal for constructing it from array source ... what was I thinking wrapping this in an iterator :/